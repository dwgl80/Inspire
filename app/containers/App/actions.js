import {
  GET_QUOTES,
  GET_QUOTES_SUCCESS,
  GET_QUOTES_FAILURE,
  SAVE_QUOTE,
  SAVE_QUOTE_SUCCESS,
  SAVE_QUOTE_FAILURE,
} from './constants';

const saveQuote = () => ({
  type: SAVE_QUOTE,
});

const saveSuccess = quote => ({
  type: SAVE_QUOTE_SUCCESS,
  quote,
});

const saveFailed = error => ({
  type: SAVE_QUOTE_FAILURE,
  error,
});

const getQuote = () => ({
  type: GET_QUOTES,
});

const quotesRetrieved = quotes => ({
  type: GET_QUOTES_SUCCESS,
  quotes,
});

const quotesFailed = error => ({
  type: GET_QUOTES_FAILURE,
  error,
});

export {
  getQuote,
  quotesRetrieved,
  quotesFailed,
  saveQuote,
  saveSuccess,
  saveFailed,
};
