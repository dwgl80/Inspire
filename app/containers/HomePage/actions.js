import { INPUT_QUOTE } from './constants';

const inputQuote = quote => ({
  type: INPUT_QUOTE,
  quote,
});

export { inputQuote };
