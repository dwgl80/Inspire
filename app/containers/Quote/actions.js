import { CHANGE_LIKED } from './constants';

const changeLiked = id => ({
  type: CHANGE_LIKED,
  id,
});

export { changeLiked };
