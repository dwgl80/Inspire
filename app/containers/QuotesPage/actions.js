import { INPUT_SEARCH } from './constants';

const inputSearch = query => ({
  type: INPUT_SEARCH,
  query,
});

export { inputSearch };
