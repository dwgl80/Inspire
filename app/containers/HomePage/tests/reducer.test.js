import { fromJS } from 'immutable';
import homePageReducer from '../reducer';
import { inputQuote } from '../actions';

describe('homePageReducer', () => {
  let testState;
  beforeEach(() => {
    testState = fromJS({
      quote: '',
    });
  });

  it('should return initial state when actions to not match', () => {
    expect(homePageReducer(undefined, {})).toMatchSnapshot();
  });
  it('should properly handle inputQuote action', () => {
    const exampleQuote = 'Shoot for the moon and the stars';
    expect(
      homePageReducer(testState, inputQuote(exampleQuote)),
    ).toMatchSnapshot();
  });
});
