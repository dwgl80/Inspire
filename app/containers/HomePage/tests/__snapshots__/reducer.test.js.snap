// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePageReducer should properly handle inputQuote action 1`] = `
Immutable.Map {
  "quote": "Shoot for the moon and the stars",
}
`;

exports[`homePageReducer should return initial state when actions to not match 1`] = `
Immutable.Map {
  "quote": "",
}
`;
